leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/serialization/EndpointContextSerDes.java

```diff
@@ -67,13 +71,12 @@ public static JsonObject serialize(EndpointContext context) {
             }
         }
         /** copy the attributes **/
-        Map<String, String> attributes = context.entries();
+        Map<String, Object> attributes = context.entries();
         if (!attributes.isEmpty()) {
-            JsonObject attContext = Json.object();
-            for (String key : attributes.keySet()) {
-                attContext.set(key, attributes.get(key));
-            }
-            peer.set(KEY_ATTRIBUTES, attContext);
+            DatagramWriter writer = new DatagramWriter();
+            SerializationUtil.write(writer, attributes);
+            String base64 = StringUtil.byteArrayToBase64(writer.toByteArray());
+            peer.set(KEY_ATTRIBUTES, base64);
```

Comment:
I'm not sure but to you break backward compatibility here ?

Reply:
The new endpointcontext uses now a typed map, not longer only string based values. Therefore the serialization was changed as well. Yes, that breaks the API and is not backwards compatible.

Reply:
Any change to make at least the deserialization backward compatible ?

Reply:
Using typed values requires either a "scheme in code" or to add the type to the serialized data. I chose the second one, because that seems to me to be most future proven. Generally, there may be solution, to add that "scheme in code" for that case.

Reply:
> Generally, there may be solution for backwards compatibility, to add that "scheme in code" for that case.

I'm not sure if it represent a lot of work ?
Currently I tried to keep backward compatibility for our distributed store. (based on redis)
I hope (not sure), I will be able to do it for the 2.0.0 final release. 

If we go with this kind of "scheme in code" (just for backward compatibility), should this put in Leshan or Californium ?

Reply:
leshan, EndpointContextSerDes. The downside is not the time of coding (at least I'm on it for a first shot), the main pain will be: we can't introduce new "mandatory" fields ;-(. (I even don't know, if that's the case at all, we will see. & months of change that lot isn't easy to remember.)